src/main/java/AuthServlet.java

```diff
@@ -0,0 +1,77 @@
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+@WebServlet(name = "AuthServlet")
+public class AuthServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String uname = request.getParameter("username");
+        String password = request.getParameter("password");
+        String generatedPassword = null;
+        try {
+            // Create MessageDigest instance for MD5
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            //Add password bytes to digest
+            md.update(password.getBytes());
+            //Get the hash's bytes
+            byte[] bytes = md.digest();
+            //This bytes[] has bytes in decimal format;
+            //Convert it to hexadecimal format
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < bytes.length; i++) {
+                sb.append(Integer.toString((bytes[i] & 0xff) + 0x100, 16).substring(1));
+            }
+            //Get complete hashed password in hex format
+            generatedPassword = sb.toString();
+        }
+        catch (NoSuchAlgorithmException e)
+        {
+            e.printStackTrace();
+        }
+
+        UserFileDaoImpl userDao = new UserFileDaoImpl();
+
+        User user = userDao.getUserByUsername(uname);
+        String destPage = "login.jsp";
+        String pass = user.getPassword();
+        String username = user.getUsername();
+
+        if (user != null && generatedPassword.equals(pass)) {
+            HttpSession session = request.getSession();
+            session.setAttribute("user", user);
+            session.setAttribute("username", username);
+            destPage = "index.jsp";
+        }
+        else {
+            String message = "Invalid username/password";
+            System.out.println("Invalid Username or Password");
+            request.setAttribute("message", message);
+        }
+
+        RequestDispatcher dispatcher = request.getRequestDispatcher(destPage);
+        dispatcher.forward(request, response);
+    }
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        HttpSession session = request.getSession(false);
+        if (session != null) {
+            session.removeAttribute("user");
+            session.removeAttribute("username");
+
+            RequestDispatcher dispatcher = request.getRequestDispatcher("login.jsp");
+            dispatcher.forward(request, response);
```

Comment:
I think here using
`sendRedirect(destPage)`
instead of lines 73 and 74 is better

Bc after successful login users can refresh freely if you use sendRedirect instead

Reply:
WAIT THESE ARE THE WRONG LINES LOL
i meant to lines **63** and **64** where u use `destPage`